teos/src/api/internal.rs

```diff
@@ -280,27 +279,40 @@ impl PrivateTowerServices for Arc<InternalAPI> {
                 .map_or("an unknown address".to_owned(), |a| a.to_string())
         );
 
-        let mut matching_appointments = vec![];
-        let locator = Locator::from_slice(&request.into_inner().locator).map_err(|_| {
+        let req_data = request.into_inner();
+        let locator = Locator::from_slice(&req_data.locator).map_err(|_| {
             Status::new(
                 Code::InvalidArgument,
                 "The provided locator does not match the expected format (16-byte hexadecimal string)",
             )
         })?;
-
-        for (_, appointment) in self
+        let mut user_id = None;
+        if let Some(user_id_slice) = req_data.user_id {
+            let parsed_user_id = UserId::from_slice(&user_id_slice).map_err(|_| {
+                Status::new(
+                    Code::InvalidArgument,
+                    "The Provided user_id does not match expected format (33-byte hex string)",
+                )
+            })?;
+            user_id = Some(parsed_user_id)
+        }
```

Comment:
What you can do instead:
```rust
let user_id = if let Some(... {
	UserId::from_slice(...
} else {
	None
};
```

Reply:
Or even better, you can do:

```suggestion
        let user_id = req_data
            .user_id
            .map(|id| UserId::from_slice(&id))
            .transpose()
            .map_err(|_| {
                Status::new(
                    Code::InvalidArgument,
                    "The Provided user_id does not match expected format (33-byte hex string)",
                )
            })?;
```